python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -928,43 +934,59 @@ def __setstate__(self, this_dict):
+        shared_dir = getattr(self_copy, name)
+
+        if hasattr(shared_dir, 'name'):
+
+            from Ganga.Core.GangaRepository import getRegistry
+            from Ganga.GPIDev.Base.Proxy import GPIProxyObjectFactory
+            shareref = GPIProxyObjectFactory(getRegistry("prep").getShareRef())
+
+            logger.debug("Increasing shareref")
+            shareref.increase(shared_dir.name)
+
     # on the deepcopy reset all non-copyable properties as defined in the
     # schema
     def __deepcopy__(self, memo=None):
         true_parent = self._getParent()
-        self = stripProxy(self)
+        raw_self = stripProxy(self)
         ## This triggers a read of the job from disk
-        self._getReadAccess()
-        self_copy = super(GangaObject, self).__deepcopy__(memo)
+        raw_self._getReadAccess()
+        classname = getName(self)
+        category = raw_self._category
+        cls = allPlugins.find(category, classname)
+
+        self_copy = cls()
 
-        if self._schema is not None:
-            for name, item in self._schema.allItems():
+        if raw_self._schema is not None:
+            for name, item in raw_self._schema.allItems():
                 if not item['copyable']:
-                    setattr(self_copy, name, self._schema.getDefaultValue(name))
-                    this_attr = getattr(self_copy, name)
-                    if isType(this_attr, Node):
-                        this_attr._setParent(self_copy)
+                    setattr(self_copy, name, raw_self._schema.getDefaultValue(name))
                 else:
-                    if hasattr(self, name):
-                        setattr(self_copy, name, deepcopy(getattr(self, name)))
+                    if hasattr(raw_self, name):
+                        setattr(self_copy, name, deepcopy(getattr(raw_self, name)))
                     else:
-                        setattr(self_copy, name, self._schema.getDefaultValue(name))
+                        setattr(self_copy, name, raw_self._schema.getDefaultValue(name))
 
+                this_attr = getattr(self_copy, name)
+                if isType(this_attr, Node):
+                    this_attr._setParent(self_copy)
 
                 if item.isA(Schema.SharedItem):
+                    self.__incrementShareRef(self_copy, name)
 
-                    shared_dir = getattr(self_copy, name)
-
-                    if hasattr(shared_dir, 'name'):
-
-                        from Ganga.Core.GangaRepository import getRegistry
-                        from Ganga.GPIDev.Base.Proxy import GPIProxyObjectFactory
-                        shareref = GPIProxyObjectFactory(getRegistry("prep").getShareRef())
+        global do_not_copy
+        for k, v in raw_self.__dict__.iteritems():
+            if k not in do_not_copy:
+                try:
+                    setattr(self_copy, k, deepcopy(v))
+                except Exception:
+                    setatttr(self_copy, k, v)
```

Comment:
Tbh I'm happy to remove the try/except here and have other classes deal with exceptions as this shouldn't be failing here (and often doesn't atm)